src/twisted/web/http.py

```diff
@@ -3254,71 +3283,84 @@ def __init__(
         self._logFormatter = logFormatter
 
         # For storing the cached log datetime and the callback to update it
-        self._logDateTime = None
-        self._logDateTimeCall = None
+        self._logDateTime: str | None = None
+        self._logDateTimeCall: IDelayedCall | None = None
+
+    logFile = property()
+
+    @logFile.getter
+    def _get_logFile(self) -> _MinimalLogFile:
+        if self._logFile is None:
+            raise AttributeError("no log file present")
+        return self._logFile
+
+    @_get_logFile.setter
+    def _set_logFile(self, newLogFile: BufferedIOBase | _MinimalLogFile) -> None:
```

Comment:
Just asking.

Why do we need all the extra code for setting a property ?

If there is a good reason, maybe this needs a comment.

```suggestion
    @property
    def logFile(self) -> _MinimalLogFile:
        if self._logFile is None:
            raise AttributeError("no log file present")
        return self._logFile

    @logFile.setter
    def logFile(self, newLogFile: BufferedIOBase | _MinimalLogFile) -> None:
```

Reply:
The code was added for a test, but as this was a public attribute, the way the test was using it (assigning a `BytesIO`) was *mostly* valid.  (It then assumed that it could read back exactly the same object, which is *not* a valid assumption, as common as that might be; if you retrieve the attribute you have to be ready to accept its actual type).

But you're right that it ought to be documented